aten/src/ATen/native/TensorShape.cpp

```diff
@@ -381,6 +381,8 @@ static inline Tensor & sparse_transpose_(Tensor & self, int64_t dim0, int64_t di
   std::vector<int64_t> sizes(self.sizes());
   std::swap(sizes[dim0], sizes[dim1]);
 
+  self._uncoalesce_();
```

Comment:
in-place transposition is defined here, so I added uncoalesce here.

Reply:
Do we need it **only** because we need to invalidate the CSR cache, not because it becomes uncoalesced? I don't think we should mark it as uncoalesced in this case, there's no reason to do so.

Reply:
We __should__ mark it uncoalesced here. Transposition of COO matrix breaks the sorted order in `indices`, so if it is kept marked coalesced, it results in incorrect behavior of matrix multiplication.  If we don't do this, the same thing as in #6219 happens. (I will add a test regarding this)